### ISSUE TITLE
[0.40] Commits to cherry-pick

### ISSUE THREAD

**Issue by mkonicek** [facebook/react-native#11423]
Cherry-picking 11 days of commits because 0.40-stable had a [huge breaking change for all iOS 3rd party libraries](https://github.com/facebook/react-native/wiki/Breaking-Changes#namespace-all-header-imports-to-react-e1577d) and this is the only way I know how to fix it:
https://github.com/facebook/react-native/commits/0.40-stable

Also cherry-picked:
- Remove dependency on Yeoman (https://github.com/facebook/react-native/commit/7ea40aa0cc4ba6e51eacf265ac9682f433cb4df9)
- Fix a bug in determining if Yarn is available (https://github.com/facebook/react-native/commit/9b534082caee0cf760aaf055583a46dd693e67e8)

Might also need:
- Fix Xcode project: https://github.com/facebook/react-native/pull/11395 (cherry picked)

**Comment by nihgwu** [comment]
@mkonicek I'm afraid that's not enough to resolve the `<React/RTCXXX.h>` migration issue, #11395 need to be cherry picked too or we can't submit the archived .ipa file

**Comment by sjmueller** [comment]
Please read my comments regarding changes in 0.40: https://github.com/facebook/react-native/commit/e1577df1fd70049ce7f288f91f6e2b18d512ff4d#commitcomment-20143360

tl;dr This still completely breaks anyone using Swift. I've tried many different configurations to unbreak our builds, but nothing helps. Please don't release a major version of RN and leave us Swift folks out in the cold!

**Comment by ncuillery** [comment]
@mkonicek I computed the diff of the template from 0.39.2 to 0.40.0-rc.3:
https://github.com/ncuillery/rn-diff/compare/rn-0.39.2...rn-0.40.0-rc.3
Maybe it could be useful for the team.

I didn't spot anything wrong in the template. But I guess some people would have a sharper eye.

**Comment by ptomasroos** [comment]
Can we release a rc 5 with #11395 any time soon @mkonicek ?

**Comment by ide** [comment]
Updated 0.40-stable with:

- 46f3ff68b649f5d7f9cc74601eca3ddb4a5cae5a (#11395 - Move headers to project section to fix archive by janicduplessis)
- 8680681a3d8e4b82240ca3aadf1b089f288f8042 (#11606 - Updated imports to libyoga*.a in HelloWorld by ptomasroos)
- bcdf00ef336d23220b427de6c1e7f259177b3693 (no PR yet - [Podspec] Make Core subspec use a Ruby string array for exclude_files by ide)

Sending to CI now.

**Comment by nihgwu** [comment]
So a new version would be cut tomorrow?

**Comment by ide** [comment]
After some changes we'll also want to pull in #11667 (CocoaPods fix for tvOS by ptomasroos) and #11676 (Update React to ~15.4.1 instead of using RC by ide).

**Comment by ptomasroos** [comment]
@ide I've changed my mind on the podspec, its not required to run RN 0.40 only to run master since this commit https://github.com/facebook/react-native/commit/c92ad5f6ae74c1d398c7cd93d5c4c50da0ca0430 is not included in the release. We run master and thats why this occurred.
